TechAndCheck/zenodotus
# Add MFA

This adds webauthn authentication as a required second factor
authentication.

To do so the following are here:
- After setting the password when setting up an account the user is
  prompted to use MFA to validate their account.
- To do the previous we also add `authenticate_user_and_setup!` to
  replace `authenticate_user!` specfically so that it validates MFA is
set up as well everywhere.
- At login the user is not logged in until after MFA is validated
- Added Lottie for animations

Before this is merged into master there will be branches off of it to do
the following:
- [x] Add recovery codes
- [x] Add TOTP for Firefox users since Firefox only implements Webauthn for
   hardware keys, not the passkey standard. This is absurd, but oh well.
  https://github.com/mdp/rotp
- [x] Add the ability to register multiple keys

To Test:
- Go through the entire account request, confirm, authorize system and
  then create a user, set a password and make sure you can register your
MFA. Do this on vault, not insights, to facilitate later testing
- Log into your account with MFA
- Log out, swtich to insights, try to log in again. Should work just
  fine
- Try it in a few different browsers with a hardware key, that should
  work. Passkey is going to be browser locked.
